src/services/account-states/account-state-engine.ts

```diff
@@ -155,11 +155,17 @@ export class AccountStateEngine {
  * Helper method to build a StateTransitionError and log relevant information when an error is to br thrown
```

Comment:
* Helper method to build a StateTransitionError and log relevant information when an error is to be thrown